File: grade_computation.c
#include <stdio.h>
#include <string.h>

float calculate_final_grade(float lab_average, float project_grade, float practical_exam_average);
void print_grade(float total_grade);

int main() {
    char repeat[4];  

    do {
        float lab_grades[3];  
        float notebook_grade, project_grade;
        float practical_exams[2];  

        for (int i = 0; i < 3; i++) {
            while (1) {
                printf("Enter the grade for Lab Report %d (0-100): ", i + 1);
                scanf("%f", &lab_grades[i]);
                if (lab_grades[i] >= 0 && lab_grades[i] <= 100) {
                    break;
                } else {
                    printf("\nInvalid input. Please enter a grade between 0 and 100.\n");
                }
            }
        }

        while (1) {
            printf("Enter the grade for the Student Notebook (0-100): ");
            scanf("%f", &notebook_grade);
            if (notebook_grade >= 0 && notebook_grade <= 100) {
                break;
            } else {
                printf("\nInvalid input. Please enter a grade between 0 and 100.\n");
            }
        }

        while (1) {
            printf("Enter the grade for the Project (0-100): ");
            scanf("%f", &project_grade);
            if (project_grade >= 0 && project_grade <= 100) {
                break;
            } else {
                printf("\nInvalid input. Please enter a grade between 0 and 100.\n");
            }
        }

        for (int i = 0; i < 2; i++) {
            while (1) {
                printf("Enter the grade for Practical Exam %d (0-100): ", i + 1);
                scanf("%f", &practical_exams[i]);
                if (practical_exams[i] >= 0 && practical_exams[i] <= 100) {
                    break;
                } else {
                    printf("\nInvalid input. Please enter a grade between 0 and 100.\n");
                }
            }
        }

        float lab_average = (lab_grades[0] + lab_grades[1] + lab_grades[2]) / 3.0;
        float practical_exam_average = (practical_exams[0] + practical_exams[1]) / 2.0;

        float total_grade = calculate_final_grade(lab_average, project_grade, practical_exam_average);

        printf("\nTotal Grade: %.2f\n", total_grade);
        print_grade(total_grade);

        printf("\nDo you want to run the program again? (yes/no): ");
        scanf("%3s", repeat);  
        while (getchar() != '\n'); 

        while (strcasecmp(repeat, "yes") != 0 && strcasecmp(repeat, "no") != 0) {
            printf("\nInvalid input. Please enter yes or no only.\n");
            printf("\nDo you want to run the program again? (yes/no): ");
            scanf("%3s", repeat);
            while (getchar() != '\n');  
        }

    } while (strcasecmp(repeat, "yes") == 0); 

    return 0;
}


float calculate_final_grade(float lab_average, float project_grade, float practical_exam_average) {
    float lab_score = lab_average * 0.40;
    float project_score = project_grade * 0.30;
    float practical_exam_score = practical_exam_average * 0.30;

    return lab_score + project_score + practical_exam_score;
}


void print_grade(float total_grade) {
    if (total_grade >= 96) {
        printf("Your Grade is: 4.0\n");
    } else if (total_grade >= 92) {
        printf("Your Grade is: 3.5\n");
    } else if (total_grade >= 88) {
        printf("Your Grade is: 3.0\n");
    } else if (total_grade >= 83) {
        printf("Your Grade is: 2.5\n");
    } else if (total_grade >= 78) {
        printf("Your Grade is: 2.0\n");
    } else if (total_grade >= 74) {
        printf("Your Grade is: 1.5\n");
    } else if (total_grade >= 70) {
        printf("Your Grade is: 1.0\n");
    } else {
        printf("Your Grade is: 0.0\n");
    }
}

File: grade_computation.py
def calculate_final_grade(lab_scores, notebook_score, project_score, practical_exams):
    lab_report_average = sum(lab_scores) / len(lab_scores)
    practical_exam_average = sum(practical_exams) / len(practical_exams)

    lab_grade = lab_report_average * 0.40
    notebook_grade = notebook_score * 0.10
    project_grade = project_score * 0.30
    practical_exam_grade = practical_exam_average * 0.20

    final_grade = lab_grade + notebook_grade + project_grade + practical_exam_grade
    
    return final_grade



def dlsu_grading_scale(final_grade):
    if final_grade > 100:
        return "Error: Total grade cannot exceed 100."
    elif 96 <= final_grade <= 100:
        return "Your DLSU Grade is: 4.0"
    elif 92 <= final_grade < 96:
        return "Your DLSU Grade is: 3.5"
    elif 88 <= final_grade < 92:
        return "Your DLSU Grade is: 3.0"
    elif 83 <= final_grade < 88:
        return "Your DLSU Grade is: 2.5"
    elif 78 <= final_grade < 83:
        return "Your DLSU Grade is: 2.0"
    elif 74 <= final_grade < 78:
        return "Your DLSU Grade is: 1.5"
    elif 70 <= final_grade < 74:
        return "Your DLSU Grade is: 1.0"
    elif final_grade < 70:
        return "Your DLSU Grade is: 0.0"
    else:
        return "Invalid Input!"



def main():
    while True:
        lab_scores = []
        
        for i in range(1, 6):
            while True:
                score = float(input(f"Enter the score for Lab Report {i} (0-100): "))
                if 0 <= score <= 100:
                    lab_scores.append(score)
                    break
                else:
                    print("\nInvalid input. Please enter a score between 0 and 100.\n")

        while True:
            notebook_score = float(input("Enter the score for the Student Notebook (0-100): "))
            if 0 <= notebook_score <= 100:
                break
            else:
                print("\nInvalid input. Please enter a score between 0 and 100.\n")

        while True:
            project_score = float(input("Enter the score for the Project (0-100): "))
            if 0 <= project_score <= 100:
                break
            else:
                print("\nInvalid input. Please enter a score between 0 and 100.\n")

        practical_exams = []
        
        for x in range(1, 3):
            while True:
                exam = float(input(f"Enter the score for Practical Exam {x} (0-100): "))
                if 0 <= exam <= 100:
                    practical_exams.append(exam)
                    break
                else:
                    print("\nInvalid input. Please enter a score between 0 and 100.\n")

        final_grade = calculate_final_grade(lab_scores, notebook_score, project_score, practical_exams)
        dlsu_grade = dlsu_grading_scale(final_grade)

        print(f"\nFinal Grade: {final_grade:.2f}")
        print(dlsu_grade)

        retry = input("\nWould you like to calculate another grade? (yes/no): ").strip().lower()
        if retry != "yes":
            print("\nThank you for using the Grade Calculator!")
            break

main()
